src/solver/conv_winoRxS_ultra_f2x3.cpp

```diff
@@ -0,0 +1,706 @@
+#else
+    std::ignore = handle;
+    std::ignore = data;
+    std::ignore = buf_ptr;
+#endif
+}
+
+} // namespace
+
+bool ConvBinWinogradUltraRxSf2x3::IsApplicable(const ExecutionContext& ctx,
+                                               const ProblemDescription& problem) const
+{
+    if(miopen::IsDisabled(MIOPEN_DEBUG_AMD_WINOGRAD_ULTRA_RXS_F2X3{}))
+        return false;
+
+#if MIOPEN_BACKEND_HIP
+    if(!problem.Is2d())
+        return false;
+    if(!problem.IsFp16())
+        return false;
+    if(!ctx.use_asm_kernels)
+        return false;
+    if(!ctx.rmv.IsV3())
+        return false;
+    if(problem.direction.IsBackwardWrW())
+        return false;
+
+    const auto name = ctx.GetStream().GetDeviceName();
+    if(!StartsWith(name, "gfx10"))
+        return false;
+
+    // clang-format off
+    if (!( problem.kernel_stride_w == 1
+        && problem.kernel_stride_w == problem.kernel_stride_h
+        && problem.kernel_dilation_w == 1
+        && problem.kernel_dilation_h == 1
+        && problem.bias == 0
+        && problem.group_counts == 1
+        && problem.in_layout == "NCHW"))
+        return false;
+    // clang-format on
+
+    const auto n_inputs_per_group  = problem.n_inputs / problem.group_counts,
+               n_outputs_per_group = problem.n_outputs / problem.group_counts;
+
+    return IsShaderContraintsMet(problem.kernel_size_h, // RxS
+                                 problem.kernel_size_w,
+                                 problem.kernel_stride_h,
+                                 problem.kernel_stride_w,
+                                 n_inputs_per_group,  // C
+                                 n_outputs_per_group, // K
+                                 problem.in_height,   // HxW
+                                 problem.in_width,
+                                 problem.out_height, // OHxOW
+                                 problem.out_width,
+                                 problem.batch_sz, // N
+                                 ctx,
+                                 problem);
+#else
+    std::ignore = ctx;
+    return false;
+#endif
+}
+
+size_t ConvBinWinogradUltraRxSf2x3::GetWorkspaceSize(const ProblemDescription& problem) const
+{
+    const auto desc          = WinogradUltraDescription(problem);
+    const size_t num_tiles_H = Ceil(desc.out_H, o_tile_H);
+    const size_t num_tiles_W = Ceil(desc.out_W, o_tile_W);
+    const size_t N           = desc.N;
+    const size_t works       = desc.flush == flush_control::FLUSH_N
+                                   ? N * Ceil(num_tiles_H * num_tiles_W, group_size)
+                                   : Ceil(N * num_tiles_H * num_tiles_W, group_size);
+
+    return works * group_size * sizeof(uint32_t);
+}
+
+ConvSolution ConvBinWinogradUltraRxSf2x3::GetSolution(const ExecutionContext& ctx,
+                                                      const ProblemDescription& problem) const

```

Comment:
All new solvers should not use legacy pattern. That massively increases the amount of later work for @averinevg with no benefit. For more info on how to correctly update the code please refer to any of the PRs like this https://github.com/ROCmSoftwarePlatform/MIOpen/pull/1821/files and feel free to ask me or Evgenii.